src/test/java/com/amazonaws/services/sqs/AmazonSQSTemporaryQueuesClientIT.java

```diff
@@ -5,61 +5,87 @@
-    }
-    
     @After
     public void teardown() {
         if (queueUrl != null) {
             client.deleteQueue(queueUrl);
         }
-        client.shutdown();
+        if (client != null) {
+            client.shutdown();
+        }
     }
     
     @Test
     public void createQueueAddsAttributes() {
-        queueUrl = client.createQueue(queueNamePrefix + "TestQueue").getQueueUrl();
-        Map<String, String> attributes = client.getQueueAttributes(queueUrl, Collections.singletonList("All")).getAttributes();
-        String hostQueueUrl = attributes.get(VIRTUAL_QUEUE_HOST_QUEUE_ATTRIBUTE);
-        assertNotNull(hostQueueUrl);
-        Assert.assertEquals("300", attributes.get(IDLE_QUEUE_RETENTION_PERIOD));
-        
-        Map<String, String> hostQueueAttributes = client.getQueueAttributes(queueUrl, Collections.singletonList("All")).getAttributes();
-        Assert.assertEquals("300", hostQueueAttributes.get(AmazonSQSIdleQueueDeletingClient.IDLE_QUEUE_RETENTION_PERIOD));
+        createQueue(null);
     }
 
     @Test
     public void createQueueWithUnsupportedAttributes() {
-        try {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            setupClient(null);
             client.createQueue(new CreateQueueRequest()
                     .withQueueName(queueNamePrefix + "InvalidQueue")
                     .withAttributes(Collections.singletonMap(QueueAttributeName.FifoQueue.name(), "true")));
-            Assert.fail("Shouldn't be able to create a FIFO temporary queue");
-        } catch (IllegalArgumentException e) {
-            Assert.assertEquals("Cannot create a temporary queue with the following attributes: FifoQueue", e.getMessage());
+        });
+    }
+
+    @Test
+    public void createQueueConfigurableIdleQueueRetentionPeriod() {
+        createQueue(200L);
+    }
+
+    @Test
+    public void createQueueWithUnsupportedIdleQueueRetentionPeriod() {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            setupClient(-10L);
+        });
+    }
+
+    private void setupClient(Long idleQueueRetentionPeriod) {
+        AmazonSQSRequesterClientBuilder requesterBuilder;
+        if (idleQueueRetentionPeriod != null) {
+            requesterBuilder =
+                    AmazonSQSRequesterClientBuilder.standard()
+                            .withAmazonSQS(sqs)
+                            .withInternalQueuePrefix(queueNamePrefix)
+                            .withQueueRetentionPeriodSeconds(idleQueueRetentionPeriod);
+        } else {
+            requesterBuilder =
+                    AmazonSQSRequesterClientBuilder.standard()
+                            .withAmazonSQS(sqs)
+                            .withInternalQueuePrefix(queueNamePrefix);
         }
+
+        client = AmazonSQSTemporaryQueuesClient.make(requesterBuilder);
+    }
+
+    private void createQueue(Long idleQueueRetentionPeriod) {
```

Comment:
nit: Consider renaming this to something more descriptive, since its not just a helper creating a queue for further testing, but actually doing the important assertions itself. Perhaps `createQueueShouldSetRetentionPeriod`?

Reply:
Renamed the function